File: PW/sch_03/plot_band_structure.jl
using PyPlot
const plt = PyPlot

function plot_band_structure( evals, kpath; filename="BANDS.pdf" )

  Nkpts = size(kpath)[2]
  Nstates = size(evals)[1]

  Xcoords = zeros(Float64,Nkpts)
  Xcoords[1] = 0.0
  for ik = 2:Nkpts
    dk = kpath[:,ik] - kpath[:,ik-1]
    Xcoords[ik] = Xcoords[ik-1] + norm( dk )
  end

  plt.clf()
  for is = 1:Nstates
    y = evals[is,:]
    plt.plot( Xcoords[:], y, marker="o")
  end
  plt.grid()
  plt.savefig(filename)
end

File: PW/sch_03/plot_bandstructure.py
from __future__ import print_function

import numpy as np
import matplotlib.pyplot as plt
import sys

plt.style.use("classic")

dat = np.loadtxt(sys.argv[1])
Nkpt = dat.shape[0]
Nstates = dat.shape[1] - 1
print("Nkpt = ", Nkpt)
print("Nstates = ", Nstates)

plt.clf()
x = dat[:,0]
for ist in range(1,Nstates+1):
    plt.plot( x, dat[:,ist], marker='o' )

filplot = sys.argv[1].replace(".dat",".pdf")
plt.savefig(filplot)

